### PR TITLE
Setup querying available appointments from DRS

### PR DESCRIPTION
**Pull request by Bhavesh-Patel**
I've attempted to separate functional logic by commits to help breakdown the overall intent of the PR.
As many files have changed, hopefully it will help to digest the changes.

### CODE REVIEW

File: HousingRepairsSchedulingApi/Startup.cs [hunk 1]
```diff
@@ -66,5 +90,46 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
...
+
+            services.Configure<DrsOptions>(drsOptionsConfiguration);
+        }
+
+        private static HttpBindingBase CreateBinding(Uri uri)
+        {
+            HttpBindingBase binding;
+            var uriScheme = uri.Scheme;
+
+            if (uriScheme == Uri.UriSchemeHttps)
```
Comment (Bhavesh-Patel) [inline HousingRepairsSchedulingApi/Startup.cs:119]: Couldn't use a `switch` here due to the schemes under `Uri` not being constants (they are `readonly` instead.)

File: HousingRepairsSchedulingApi/Gateways/DrsAppointmentGateway.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            Guard.Against.NullOrWhiteSpace(sorCode, nameof(sorCode));
+            Guard.Against.NullOrWhiteSpace(locationId, nameof(locationId));
+
+            var earliestDate = fromDate ?? DateTime.Today.AddDays(appointmentLeadTimeInDays);
+            var drsAppointmentSlots = Enumerable.Empty<DrsAppointmentSlot>();
+
+            while (drsAppointmentSlots.Select(x => x.StartTime.Date).Distinct().Count() < requiredNumberOfAppointmentDays)
+            {
+                var appointments = await drsService.CheckAvailability(sorCode, locationId, earliestDate);
+                appointments = appointments.Where(x =>
```
Comment (maysakanoni) [inline HousingRepairsSchedulingApi/Gateways/DrsAppointmentGateway.cs:44]: Are we including or excluding appointments at 9.30 and 2.30?

File: HousingRepairsSchedulingApi/Gateways/DrsAppointmentGateway.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            Guard.Against.NegativeOrZero(appointmentSearchTimeSpanInDays, nameof(appointmentSearchTimeSpanInDays));
+            Guard.Against.Negative(appointmentLeadTimeInDays, nameof(appointmentLeadTimeInDays));
+
+            this.drsService = drsService;
+            this.requiredNumberOfAppointmentDays = requiredNumberOfAppointmentDays;
+            this.appointmentSearchTimeSpanInDays = appointmentSearchTimeSpanInDays;
+            this.appointmentLeadTimeInDays = appointmentLeadTimeInDays;
+        }
+
+        public async Task<IEnumerable<AppointmentSlot>> GetAvailableAppointments(string sorCode, string locationId,
```
Comment (maysakanoni) [inline HousingRepairsSchedulingApi/Gateways/DrsAppointmentGateway.cs:32]: Is it appropriate to make this return a HACT object or do we want to decouple the domain of the application from the HACT data standard, and if so why?

File: HousingRepairsSchedulingApi/Gateways/DrsAppointmentGateway.cs (hunk 2 above)
Comment (Bhavesh-Patel) [reply HousingRepairsSchedulingApi/Gateways/DrsAppointmentGateway.cs:44]: The 0930 - 1430 slot is being excluded by decision of the PO (possibly after user research).

File: HousingRepairsSchedulingApi/Gateways/DrsAppointmentGateway.cs (hunk 3 above)
Comment (Bhavesh-Patel) [reply HousingRepairsSchedulingApi/Gateways/DrsAppointmentGateway.cs:32]: I wanted to keep the gateway decoupled from the domain to create another layer of abstraction between HACT, application and 3rd party service (DRS).

Once implemented, I was unsure if it was necessary.

Happy to remove `AppointmentSlot` and restore to using the HACT standard throughout and having the gateway convert the service response to HACT.

Comment (Bhavesh-Patel) [reply HousingRepairsSchedulingApi/Gateways/DrsAppointmentGateway.cs:31]: As discussed, removed extra, unnecessary domain layer.

**Review by elena-vi — APPROVED** [review]

**Comment by Bhavesh-Patel** [comment]
[Integrate scheduling API with external scheduling system to query availability](https://trello.com/c/uFoyO6GJ/68-integrate-scheduling-api-with-external-scheduling-system-to-query-availability)

**Pull request merged**